src/ucp/rndv/rndv_ppln.c

```diff
@@ -179,11 +179,25 @@ void ucp_proto_rndv_ppln_send_frag_complete(ucp_request_t *freq, int send_ack)
                                       "ppln_send");
 }
 
+static ucs_status_t ucp_proto_rndv_ppln_recv_frag_reset(ucp_request_t *req)
+{
+    ucs_assert(req->send.rndv.ppln.ack_data_size == 0);
+
+    if (!ucp_proto_common_frag_is_completed(req)) {
```

Comment:
ucp_proto_common_frag_is_completed is actually expecting a top-level request (not a frag request) since it's checking completed_size field. So its name is misleading. It should be something like ucp_proto_common_multi_frag_is_completed
here also, ucp_proto_rndv_ppln_recv_frag_reset is expecting a top-level request, so its name should be ucp_proto_rndv_recv_ppln_reset or ucp_proto_rndv_recv_ppln_clean